scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -1814,6 +1858,12 @@ def _get_containers(
                                             ignore_k8s_api_exception=True,
                                         )
                                         if pod:
+                                            # We've read the pod from the cache, so any WarmingEntry associated
+                                            # with this is no longer new.  This ensures we pick up short lived
+                                            # logs that finished before they had a warm entry in the k8s_cache
+                                            if controlled_warmer is not None:
+                                                controlled_warmer.mark_not_new(cid)
```

Comment:
I would also propose we add in:

```
   if is_old_dead_container:
       controlled_warmer.unmark_to_warm(cid, namespace, pod_name)
```

The main reason is that my head hurts that when we finish marking in this iteration, old dead containers will still in the active set.  I think that would be surprising ... even though with how the code currently is written, it doesn't make a difference.  However, who knows how we could change this in the future, so I think it's better to just make is so that the dead container doesn't get included.

Reply:
Added a parameter to `mark_has_been_used` to say whether or not to also unmark from the warmer.